File: frontend/src/types/collect.ts
export interface CollectResult {
  batch_id: string
  source_id: number
  source_name: string
  record_count: number
  status: string
  message: string
}

export interface CollectHistoryItem {
  batch_id: string
  source_id: number
  source_name: string
  source_type: string
  record_count: number
  status: string
  created_at: string
}

export interface BatchDetail {
  batch_id: string
  source_id: number
  source_name: string
  source_type: string
  total_records: number
  status: string
  raw_records: Array<{
    id: number
    raw_data: Record<string, any>
    source_row_index: number
  }>
  created_at: string
}

File: app/schemas/collect.py
from datetime import datetime
from typing import Any, Optional
from pydantic import BaseModel, Field


class CollectResponse(BaseModel):
    """单源采集响应。"""

    batch_id: str
    source_id: int
    source_name: str
    record_count: int
    status: str
    message: str = ""


class BatchCollectRequest(BaseModel):
    """批量采集请求。"""

    source_ids: list[int] = Field(..., min_length=1, max_length=50)


class BatchCollectResponse(BaseModel):
    """批量采集响应。"""

    total_sources: int
    success_count: int
    failed_count: int
    results: list[CollectResponse]


class CollectHistoryItem(BaseModel):
    """采集历史项。"""

    batch_id: str
    source_id: int
    source_name: str
    source_type: str
    record_count: int
    status: str
    created_at: datetime


class BatchDetailResponse(BaseModel):
    """批次详情（含原始数据预览）。"""

    batch_id: str
    source_id: int
    source_name: str
    source_type: str
    total_records: int
    status: str
    raw_records: list[dict[str, Any]]  # 前 100 条预览
    created_at: datetime
